### ISSUE TITLE
Build image failed with Permission denied

### ISSUE THREAD

**Issue by r0bj** [GoogleContainerTools/kaniko#1614]
**Actual behavior**
Build image is failing with "Permission denied" during installing python dependencies.

**Expected behavior**
Image is build without any issue.
The same image can build successfully using docker.

**To Reproduce**
1. Context directory:
```
Dockerfile
requirements.txt
```
Dockerfile:
```
FROM python:3.7.2-alpine3.8
RUN addgroup -g 101 -S python && adduser -u 101 -S python -G python; apk add --no-cache --virtual .build-deps gcc musl-dev curl && chown -R 101:101 /home/python
USER 101
WORKDIR /home/python
COPY . .
RUN pip install --no-cache-dir -r requirements.txt -t .
```
requirements.txt:
```
aiobotocore==1.1.2
coloredlogs==14.0
```

2. Build image with kaniko:
```
docker run -v "$HOME"/image_context:/workspace gcr.io/kaniko-project/executor:v1.5.2 --dockerfile /workspace/Dockerfile --context dir:///workspace/ --no-push
```

Error message is generated:
```
PermissionError: [Errno 13] Permission denied: '/tmp/pip-target-nz8165ys/lib/python/attrs-20.3.0.dist-info' -> '/home/python/attrs-20.3.0.dist-info'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/pip/_internal/cli/base_command.py", line 179, in main
    status = self.run(options, args)
  File "/usr/local/lib/python3.7/site-packages/pip/_internal/commands/install.py", line 441, in run
    options.target_dir, target_temp_dir, options.upgrade
  File "/usr/local/lib/python3.7/site-packages/pip/_internal/commands/install.py", line 498, in _handle_target_dir
    target_item_dir
  File "/usr/local/lib/python3.7/shutil.py", line 574, in move
    symlinks=True)
  File "/usr/local/lib/python3.7/shutil.py", line 321, in copytree
    os.makedirs(dst)
  File "/usr/local/lib/python3.7/os.py", line 221, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/home/python/attrs-20.3.0.dist-info'
```
 
 **Triage Notes for the Maintainers**
 

 
 | **Description** | **Yes/No** |
 |----------------|---------------|
 | Please check if this a new feature you are proposing        | <ul><li>- [ ] </li></ul>|
 | Please check if the build works in docker but not in kaniko | <ul><li>- [x] </li></ul>| 
 | Please check if this error is seen when you use `--cache` flag | <ul><li>- [x] </li></ul>|
 | Please check if your dockerfile is a multistage dockerfile | <ul><li>- [ ] </li></ul>|

**Comment by aaron-prindle** [comment]
Doing some poking around, I think that the `chown` command is possibly not working as expected in this case, not sure the root cause.

When  I do the following:

`Dockerfile`
```
FROM python:3.7.2-alpine3.8
RUN addgroup -g 101 -S python && adduser -u 101 -S python -G python; apk add --no-cache --virtual .build-deps gcc musl-dev curl && chown -R 101:101 /home/python
USER 101
WORKDIR /home/python
COPY . .
RUN sh -c "while :; do echo 'just looping here... nothing special'; sleep 1; done"

# RUN pip install --no-cache-dir -r requirements.txt -t .
```

run Kaniko and such  that it hangs before failing command
```
aprindle@aprindle-ssd ~/kaniko  [aprindle-debug]./run_in_docker.sh /workspace/Dockerfile /usr/local/google/home/aprindle/kaniko-ctx-dir gcr.io/aprindle-test-cluster/kaniko-test:latest false |& tee ~/out.txt
...
just looping here... nothing special
just looping here... nothing special
```

look at /home/python ownership/perms
```
aprindle@aprindle-ssd ~/dockerfile-loop docker exec -it 80bb8026681b sh
/workspace # 
/workspace # ls
Dockerfile        requirements.txt
/workspace # ls -l
total 8
-rw-r--r--    1 386724   89939          369 Jul 14 23:37 Dockerfile
-rw-r--r--    1 386724   89939           36 Jul 14 23:34 requirements.txt
```

it is not owned by `python` group/101 but a different UID/GID

**Comment by atlasloewenherz** [comment]
this seem to be a very puzzeling issue we are facing it within a gitlab runner's pipeline using multi stage build and it manifests it self as following:

TLDR:
`INFO[0176] Taking snapshot of full filesystem...        
INFO[0187] Saving file opt/pysetup/.venv for later use  
error building image: could not save file: copying file: symlink /usr/local/bin/python /kaniko/1/opt/pysetup/.venv/bin/python: file exists
Cleaning up project directory and file based variables
00:04
ERROR: Job failed: exit code 1`

or the full logs:

[pipeline.log](https://github.com/GoogleContainerTools/kaniko/files/12285233/pipeline.log)

and this is the [Dockerfile.prod.txt](https://github.com/GoogleContainerTools/kaniko/files/12285255/Dockerfile.prod.txt)

i'm not sure if this help get a clear context or just make it even more confusing!

Thanks & ill be happy to provide more logs/info if required
